Fixed Warnings

vigbalu opened the pull request:
Fixed few warnings in ompd and gdb-plugin code.

vigbalu committed:
Fixed Warnings

jinisusan reviewed (approved):
LGTM

jprotze commented on openmp/libompd/gdb-plugin/ompdModule.c:
```diff
@@ -574,7 +574,7 @@ ompd_rc_t _read_string (
 	Py_ssize_t retSize;
 	const char* strbuffer = PyUnicode_AsUTF8AndSize(retString, &retSize);
-	if(retSize >= nbytes) {
+	if((unsigned int)retSize >= nbytes) {
```
> The returned buffer always has an extra null byte appended (not included in size), regardless of whether there are any other null code points.
I think this should be:
```suggestion
	if((ompd_size_t)retSize+1 >= nbytes) {
```

jprotze commented on openmp/libompd/gdb-plugin/ompdModule.c:
```diff
@@ -542,7 +542,7 @@ ompd_rc_t _read (
 		Py_ssize_t retSize = PyByteArray_Size(retArray);
 		const char* strBuf = PyByteArray_AsString(retArray);
-		if(retSize != nbytes) {
+		if((unsigned int)retSize != nbytes) {
```
```suggestion
		if((ompd_size_t)retSize != nbytes) {
```

vigbalu committed:
Fixing API's return code to be valid asper std doc

1, "ompd_initialize" should be called with valid OMPD "version", as of now supported libompd version is "201811(OMPD_VERSION)"
2, Other minor changes to check sanity for arguments to return proper return code.

vigbalu commented:
Created a new pull request #38 with review comments addressed.
